webapp/app/js/compareperf.js

```diff
@@ -311,14 +345,24 @@ perf.controller('CompareSubtestResultsCtrl', [
               }).then(function(seriesMaps) {
                   var originalSeriesMap = seriesMaps[$scope.originalResultSetID];
                   var newSeriesMap = seriesMaps[$scope.newResultSetID];
-                  Object.keys(originalSeriesMap).forEach(function(series) {
-                    if (!_.contains($scope.pageList, originalSeriesMap[series].name)) {
-                      $scope.pageList.push(originalSeriesMap[series].name);
-                    }
-                  });
+                  if (originalSeriesMap) {
+                    Object.keys(originalSeriesMap).forEach(function(series) {
+                      if (!_.contains($scope.pageList, originalSeriesMap[series].name)) {
+                        $scope.pageList.push(originalSeriesMap[series].name);
+                      }
+                    });
+                  }
+                  if (newSeriesMap) {
+                    Object.keys(newSeriesMap).forEach(function(series) {
+                      if (!_.contains($scope.pageList, newSeriesMap[series].name)) {
+                        $scope.pageList.push(newSeriesMap[series].name);
+                      }
+                    });
+                  }
```

Comment:
You could put this logic in a forEach that iterates over originalSeriesMap and newSeriesMap